string type # stream type
uint32 uid  # stream uid in database